TEST: 2.2.x + blas variants

h-vetinari opened the pull request:
Continuing the analysis from #326, #317, #307, #293, #288, #273, #252, #237, #227 & #196. Do not merge.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-recipe-manager`. Your recipe may not receive automatic updates and/or may not be compatible with conda-forge's infrastructure. Please check the logs for more information and ensure your recipe can be parsed.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/12248515405](https://github.com/conda-forge/conda-forge-webservices/actions/runs/12248515405). Examine the logs at this URL for more detail.</sub>

h-vetinari commented:
### Update for numpy 2.2.0

From all green [2.1.0](https://github.com/conda-forge/numpy-feedstock/pull/326#issuecomment-2295677351) (but actually 3/51 failures due to a latent test skip; see below), we're now in the same situation 5/85 failures, i.e. only linux + aarch64 + OpenBLAS fails.

### Notable

- Added py3.13 and py3.13t
- Removed some old test skips, one successfully, one unsuccessfully
- In particular, the skip added in https://github.com/conda-forge/numpy-feedstock/commit/4db75ebf54ca4de33d73a4ee4c6bbe617919e540 is still necessary (this was added in the 1.26.x timeframe and so this failure is not strictly speaking new)
- the following test skips remained in place:
```yaml
# there are some tests that cannot really work in emulation, see e.g. numpy/numpy#20445
{% set tests_to_skip = tests_to_skip + " or Test_ARM_Features" %}                   # [build_platform != target_platform]
{% set tests_to_skip = tests_to_skip + " or Test_POWER_Features" %}                 # [build_platform != target_platform]
# test_new_policy reruns part of test suite; including a CPU feature test that fails in emulation
{% set tests_to_skip = tests_to_skip + " or test_new_policy" %}                     # [build_platform != target_platform]
```

### State of BLAS/LAPACK

- Continued problems with [bringing up](https://github.com/conda-forge/blas-feedstock/pull/114) non-ancient LAPACK, now also [problems](https://github.com/conda-forge/intel_repack-feedstock/issues/83) with MKL 2025.0
- Flang migration is still waiting for blas due to the above-mentioned issues
- Blis 1.0 is [stuck](https://github.com/conda-forge/blis-feedstock/pull/35) too


### Details

| lib | [for 2.1.0](https://github.com/conda-forge/numpy-feedstock/pull/326#issuecomment-2295677351) | after | updated<br/>version | updated<br/>build |
|--------|-------|----|-|-|
| `numpy` | `2.1.0` | `2.2.0` | X | |
| `libblas` | `3.9.0-23` | `3.9.0-25` | | X |
| `blis` | `0.9.0-2` | `0.9.0-2` | | |
| `openblas` | `0.3.27-pthreads-0` | `0.3.28-pthreads-1` | X | |
| `mkl` | `2024.2.1-103` (linux)<br/>`2023.2.0-50500` (osx)<br/>`2024.1.0-694` (win) | `2024.2.2-16` (linux)<br/>`2023.2.0-50500` (osx)<br/>`2024.2.2-14` (win) | X | |
| `netlib` | `3.9.0-6` | `3.9.0-8` | | X |
| [`qemu-user-static`](https://github.com/conda-forge/docker-images/blob/main/download-qemu-static.sh#L24) | `8.2.4-1` | `9.1.2-2` | X | |


--------------------

| variant | [2.1.0](https://github.com/conda-forge/numpy-feedstock/pull/326#issuecomment-2295677351) | now |
|--------|-------|----|
| linux + aarch64 + openblas | 4 failures in `TestCholesky::test_basic_property`,<br/>apparently due to emulation | remain |

---------------

| variant | accelerate | blis | mkl | netlib | openblas | sum** | 
|--------|------------|----|-----|-------|-----------|-------|
| linux / x86 | ➖ | ✔️ | ✔️ | ✔️ | ✔️ | - | 
| linux / aarch | ➖ | ➖ | ➖ | ✔️ | ❌  | 5F | 
| linux / ppc64le | ➖ | ➖ | ➖ | ✔️ | ✔️ | - |
| osx / arm | ❔ | ➖ | ➖ | ❔ | ❔ | - |
| osx / x86 | ✔️ | ✔️ | ✔️ | ✔️ | ✔️ | - |
| win / x86 | ➖ | ✔️ | ✔️ | ✔️ | ✔️| - |
| sum** | - | - | - | - | 5F | 5F |

** sum of Segfaults (S), Failures (F), resp. Timeouts (T); out of a total of 100 CI combinations being tested

Build logs:
[Azure](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1104502&view=results)

---------------

<details>
<summary>linux + aarch64 + openblas: 4 test failures</summary>

```
=========================== short test summary info ============================
FAILED linalg/tests/test_linalg.py::TestCholesky::test_basic_property[False-float32-shape3] - AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0.00298023
(50, 50) <class 'numpy.float32'>
[[ 44.06959     -1.2917366   -0.61489445 ...  -1.7078063   -0.06026965
    3.5968263 ]
 [ -1.2917366   50.031647    -0.93625164 ...  -9.246351   -12.977635
   -0.22433573]
 [ -0.61489445  -0.93625164  56.479916   ...   3.1758978    6.96665
   -2.8806841 ]
 ...
 [ -1.7078063   -9.246351     3.1758978  ...  53.048283     6.9221125
    6.9285684 ]
 [ -0.06026965 -12.977635     6.96665    ...   6.9221125   49.258278
  -11.654396  ]
 [  3.5968263   -0.22433573  -2.8806841  ...   6.9285684  -11.654396
   47.500088  ]]
[[ 6.638493    0.          0.         ...  0.          0.
   0.        ]
 [-0.1945828   7.070628    0.         ...  0.          0.
   0.        ]
 [-0.0926256  -0.13496326  7.5135293  ...  0.          0.
   0.        ]
 ...
 [-0.25725812 -1.3147925   0.3959019  ...  0.7488741   0.
   0.        ]
 [-0.00907881 -1.8356787   0.8941284  ... -0.11000964  0.70787036
   0.        ]
 [ 0.5418137  -0.01681719 -0.3770223  ... -0.01939702  0.88379747
   1.6186445 ]]
Mismatched elements: 1617 / 2500 (64.7%)
Max absolute difference among violations: 89.25262
Max relative difference among violations: 2200.9631
 ACTUAL: array([[ 4.406959e+01, -1.291737e+00, -6.148944e-01, ..., -9.883090e+00,
        -2.005572e+00,  7.965497e+00],
       [-1.291737e+00, -7.090253e+00,  1.104174e+00, ..., -9.404308e-03,...
 DESIRED: array([[ 44.06959 ,  -1.291737,  -0.614894, ...,  -1.707806,  -0.06027 ,
          3.596826],
       [ -1.291737,  50.031647,  -0.936252, ...,  -9.246351, -12.977635,...
FAILED linalg/tests/test_linalg.py::TestCholesky::test_basic_property[False-float32-shape4] - AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0.000178814
(3, 10, 10) <class 'numpy.float32'>
[[[10.734333   -4.8768277  -3.1554568  -4.5997534   2.1030364
   -1.2435193   3.0306861  -4.149522   -2.9620435   0.03625464]
  [-4.8768277   8.896948    3.812659   -0.03711339 -1.0476841
    4.170913    0.7686802   0.3364729   1.0283238  -3.826606  ]
  [-3.1554568   3.812659    3.9959366   1.1625074   0.8306916
    0.39898828 -0.77481    -1.9865611  -0.5670059  -2.2059839 ]
  [-4.5997534  -0.03711339  1.1625074   6.6847258  -0.24326262
   -0.47384936 -1.5542715   4.114234   -1.7963581   1.5196537 ]
  [ 2.1030364  -1.0476841   0.8306916  -0.24326262  3.89964
   -4.6199813   1.9657652  -1.6450504  -0.18314792  0.349001  ]
  [-1.2435193   4.170913    0.39898828 -0.47384936 -4.6199813
   11.839529   -2.9300923   0.8218101  -1.1914967  -1.5600258 ]
  [ 3.0306861   0.7686802  -0.77481    -1.5542715   1.9657652
   -2.9300923   7.020295    1.0565685   1.5246283  -3.3805852 ]
  [-4.149522    0.3364729  -1.9865611   4.114234   -1.6450504
    0.8218101   1.0565685  10.698549    4.4092803   0.4416819 ]
  [-2.9620435   1.0283238  -0.5670059  -1.7963581  -0.18314792
   -1.1914967   1.5246283   4.4092803   8.2551985  -1.8104365 ]
  [ 0.03625464 -3.826606   -2.2059839   1.5196537   0.349001
   -1.5600258  -3.3805852   0.4416819  -1.8104365   6.6920257 ]]

 [[ 5.9013734  -0.11575395 -1.8299209  -0.09370685  0.29582188
    0.94792557 -1.7730689   3.0220997  -2.5740616   2.1073136 ]
  [-0.11575395 10.544705    0.79119295 -5.9150043  -1.6428163
   -1.3827156   0.25873902 -1.2621315  -0.7767093   4.0993524 ]
  [-1.8299209   0.79119295 16.183662    6.329022   -1.1351309
   -2.710929    6.666974    0.74811184  2.4979637   1.5011857 ]
  [-0.09370685 -5.9150043   6.329022   13.666377   -2.3753624
    0.70267767  4.4945393   1.8476337  -2.3819492   1.1581259 ]
  [ 0.29582188 -1.6428163  -1.1351309  -2.3753624   6.571176
   -4.508656   -2.573465   -0.43898493  5.2973313  -1.8500149 ]
  [ 0.94792557 -1.3827156  -2.710929    0.70267767 -4.508656
    6.8080935   0.086738    2.382368   -5.498594   -0.53745675]
  [-1.7730689   0.25873902  6.666974    4.4945393  -2.573465
    0.086738    6.919965   -0.40823123  1.9532615  -0.06429972]
  [ 3.0220997  -1.2621315   0.74811184  1.8476337  -0.43898493
    2.382368   -0.40823123  5.7033362  -2.0585406   0.21069823]
  [-2.5740616  -0.7767093   2.4979637  -2.3819492   5.2973313
   -5.498594    1.9532615  -2.0585406   9.462108   -3.9543934 ]
  [ 2.1073136   4.0993524   1.5011857   1.1581259  -1.8500149
   -0.53745675 -0.06429972  0.21069823 -3.9543934   7.435262  ]]

 [[ 4.6643376   2.6854048   4.0593524   0.1535468  -1.5118827
    3.397957   -2.9920995   0.80777925 -1.7318133   0.82475716]
  [ 2.6854048   7.829547   -0.5087376   4.3746657  -1.4713306
   -4.861691   -6.0652037   0.99772763 -4.4356337   5.25942   ]
  [ 4.0593524  -0.5087376  13.68554    -5.680109   -4.2268295
    8.056314   -6.3996224  -0.8219479  -1.7537458   1.7455587 ]
  [ 0.1535468   4.3746657  -5.680109   15.329586   -1.0279889
   -9.684909   -4.311326   -0.32577878 -8.493642    7.0410876 ]
  [-1.5118827  -1.4713306  -4.2268295  -1.0279889   6.4806786
   -1.5806129   3.722421    1.541739   -0.32345185 -3.0850477 ]
  [ 3.397957   -4.861691    8.056314   -9.684909   -1.5806129
   15.264503    3.3202677  -0.3396074   4.926388   -7.346789  ]
  [-2.9920995  -6.0652037  -6.3996224  -4.311326    3.722421
    3.3202677  11.730729    1.6513809   7.1151867  -8.045567  ]
  [ 0.80777925  0.99772763 -0.8219479  -0.32577878  1.541739
   -0.3396074   1.6513809   4.239041   -0.10079275 -0.07716311]
  [-1.7318133  -4.4356337  -1.7537458  -8.493642   -0.32345185
    4.926388    7.1151867  -0.10079275 12.2900095  -7.07966   ]
  [ 0.82475716  5.25942     1.7455587   7.0410876  -3.0850477
   -7.346789   -8.045567   -0.07716311 -7.07966     9.485415  ]]]
[[[ 3.2763293   0.          0.          0.          0.
    0.          0.          0.          0.          0.        ]
  [-1.4885036   2.584822    0.          0.          0.
    0.          0.          0.          0.          0.        ]
  [-0.96310735  0.9204      1.4903773   0.          0.
    0.          0.          0.          0.          0.        ]
  [-1.4039351  -0.8228325   0.38091183  1.9726999   0.
    0.          0.          0.          0.          0.        ]
  [ 0.641888   -0.03568195  0.9942048   0.12664968  1.5753931
    0.          0.          0.          0.          0.        ]
  [-0.3795465   1.3950504  -0.83908963  0.23359051 -2.2355907
    1.9982008   0.          0.          0.          0.        ]
  [ 0.9250249   0.83006996 -0.43472758  0.3006057   1.1398793
   -0.8125721   1.7990714   0.          0.          0.        ]
  [-1.2665155  -0.599167   -1.7813463   1.2782736   0.4796653
    0.22821037  0.67006654  1.7880605   0.          0.        ]
  [-0.90407383 -0.12279119 -0.8888414  -1.4336116   0.9255097
    0.14752597  0.87395006  1.3292007   1.0810157   0.        ]
  [ 0.01106563 -1.4740415  -0.5626888   0.27203056  0.5168721
    0.5606847  -1.4603251  -0.657102   -0.46530387  0.87528366]]

 [[ 2.4292743   0.          0.          0.          0.
    0.          0.          0.          0.          0.        ]
  [-0.0476496   3.2469115   0.          0.          0.
    0.          0.          0.          0.          0.        ]
  [-0.75327885  0.23262091  3.9448855   0.          0.
    0.          0.          0.          0.          0.        ]
  [-0.03857401 -1.8222985   1.7044525   2.7274456   0.
    0.          0.          0.          0.          0.        ]
  [ 0.12177376 -0.50417566 -0.2347646  -1.059335    2.263813
    0.          0.          0.          0.          0.        ]
  [ 0.39020938 -0.42012918 -0.5879161   0.3498526  -2.0034354
    1.4133387   0.          0.          0.          0.        ]
  [-0.729876    0.06897654  1.5465921   0.7171502  -0.58618873
   -0.08172157  1.768033    0.          0.          0.        ]
  [ 1.2440339  -0.37046087  0.4490354   0.16688555 -0.2186787
    1.067539   -0.18652947  1.6021042   0.          0.        ]
  [-1.0596011  -0.25476483  0.44590706 -1.3371879   1.7607771
   -0.6612598   1.3828325   0.3352417   0.8516632   0.        ]
  [ 0.86746633  1.2752694   0.47098336  0.9946084  -0.06559554
   -0.3839536  -0.58293706 -0.30378163 -0.96378136  1.5273129 ]]

 [[ 2.1597078   0.          0.          0.          0.
    0.          0.          0.          0.          0.        ]
  [ 1.2434112   2.5066862   0.          0.          0.
    0.          0.          0.          0.          0.        ]
  [ 1.8795841  -1.1352971   2.977214    0.          0.
    0.          0.          0.          0.          0.        ]
  [ 0.0710961   1.7099324  -1.3006988   3.272437    0.
    0.          0.          0.          0.          0.        ]
  [-0.7000404  -0.23971593 -1.0691853  -0.5986391   2.1051443
    0.          0.          0.          0.          0.        ]
  [ 1.5733411  -2.719926    0.6755185  -1.3039918  -0.5650853
    1.707355    0.          0.          0.          0.        ]
  [-1.3854187  -1.7323902  -1.9354969  -1.1514527  -0.20018531
    0.2816568   1.2723097   0.          0.          0.        ]
  [ 0.37402248  0.21249725 -0.43117726 -0.39009395  0.5510192
   -0.1500179   1.1054928   1.4722989   0.          0.        ]
  [-0.8018739  -1.3717611  -0.60590625 -2.102138   -1.4820251
   -0.41726547 -0.11362489  0.19627525  1.5991338   0.        ]
  [ 0.38188368  1.9087278   1.0730667   1.572491   -0.12896962
   -0.8804634  -0.07867548  0.32361656 -0.51923645  0.9485597 ]]]
Mismatched elements: 253 / 300 (84.3%)
Max absolute difference among violations: 20.29966
Max relative difference among violations: 237.33156
 ACTUAL: array([[[10.734334, -4.876828, -3.155457, -4.599753,  2.103036,
         -1.243519,  3.030686, -4.149522,  0.      ,  0.      ],
        [-4.876828,  2.244246,  1.433589,  8.771067,  1.423618,...
 DESIRED: array([[[10.734333, -4.876828, -3.155457, -4.599753,  2.103036,
         -1.243519,  3.030686, -4.149522, -2.962044,  0.036255],
        [-4.876828,  8.896948,  3.812659, -0.037113, -1.047684,...
FAILED linalg/tests/test_linalg.py::TestCholesky::test_basic_property[True-float32-shape3] - AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0.00298023
(50, 50) <class 'numpy.float32'>
[[ 44.06959     -1.2917366   -0.61489445 ...  -1.7078063   -0.06026965
    3.5968263 ]
 [ -1.2917366   50.031647    -0.93625164 ...  -9.246351   -12.977635
   -0.22433573]
 [ -0.61489445  -0.93625164  56.479916   ...   3.1758978    6.96665
   -2.8806841 ]
 ...
 [ -1.7078063   -9.246351     3.1758978  ...  53.048283     6.9221125
    6.9285684 ]
 [ -0.06026965 -12.977635     6.96665    ...   6.9221125   49.258278
  -11.654396  ]
 [  3.5968263   -0.22433573  -2.8806841  ...   6.9285684  -11.654396
   47.500088  ]]
[[ 6.638493   -0.1945828  -0.0926256  ... -0.25725812 -0.00907881
   0.5418137 ]
 [ 0.          7.070628   -0.13496326 ... -1.3147925  -1.8356787
  -0.01681719]
 [ 0.          0.          7.5135293  ...  0.3959019   0.8941284
  -0.3770223 ]
 ...
 [ 0.          0.          0.         ...  0.7488741  -0.11000964
  -0.01939702]
 [ 0.          0.          0.         ...  0.          0.70787036
   0.88379747]
 [ 0.          0.          0.         ...  0.          0.
   1.6186445 ]]
Mismatched elements: 1617 / 2500 (64.7%)
Max absolute difference among violations: 89.25262
Max relative difference among violations: 2200.9631
 ACTUAL: array([[ 4.406959e+01, -1.291737e+00, -6.148944e-01, ..., -9.883090e+00,
        -2.005572e+00,  7.965497e+00],
       [-1.291737e+00, -7.090253e+00,  1.104174e+00, ..., -9.404308e-03,...
 DESIRED: array([[ 44.06959 ,  -1.291737,  -0.614894, ...,  -1.707806,  -0.06027 ,
          3.596826],
       [ -1.291737,  50.031647,  -0.936252, ...,  -9.246351, -12.977635,...
FAILED linalg/tests/test_linalg.py::TestCholesky::test_basic_property[True-float32-shape4] - AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0.000178814
(3, 10, 10) <class 'numpy.float32'>
[[[10.734333   -4.8768277  -3.1554568  -4.5997534   2.1030364
   -1.2435193   3.0306861  -4.149522   -2.9620435   0.03625464]
  [-4.8768277   8.896948    3.812659   -0.03711339 -1.0476841
    4.170913    0.7686802   0.3364729   1.0283238  -3.826606  ]
  [-3.1554568   3.812659    3.9959366   1.1625074   0.8306916
    0.39898828 -0.77481    -1.9865611  -0.5670059  -2.2059839 ]
  [-4.5997534  -0.03711339  1.1625074   6.6847258  -0.24326262
   -0.47384936 -1.5542715   4.114234   -1.7963581   1.5196537 ]
  [ 2.1030364  -1.0476841   0.8306916  -0.24326262  3.89964
   -4.6199813   1.9657652  -1.6450504  -0.18314792  0.349001  ]
  [-1.2435193   4.170913    0.39898828 -0.47384936 -4.6199813
   11.839529   -2.9300923   0.8218101  -1.1914967  -1.5600258 ]
  [ 3.0306861   0.7686802  -0.77481    -1.5542715   1.9657652
   -2.9300923   7.020295    1.0565685   1.5246283  -3.3805852 ]
  [-4.149522    0.3364729  -1.9865611   4.114234   -1.6450504
    0.8218101   1.0565685  10.698549    4.4092803   0.4416819 ]
  [-2.9620435   1.0283238  -0.5670059  -1.7963581  -0.18314792
   -1.1914967   1.5246283   4.4092803   8.2551985  -1.8104365 ]
  [ 0.03625464 -3.826606   -2.2059839   1.5196537   0.349001
   -1.5600258  -3.3805852   0.4416819  -1.8104365   6.6920257 ]]

 [[ 5.9013734  -0.11575395 -1.8299209  -0.09370685  0.29582188
    0.94792557 -1.7730689   3.0220997  -2.5740616   2.1073136 ]
  [-0.11575395 10.544705    0.79119295 -5.9150043  -1.6428163
   -1.3827156   0.25873902 -1.2621315  -0.7767093   4.0993524 ]
  [-1.8299209   0.79119295 16.183662    6.329022   -1.1351309
   -2.710929    6.666974    0.74811184  2.4979637   1.5011857 ]
  [-0.09370685 -5.9150043   6.329022   13.666377   -2.3753624
    0.70267767  4.4945393   1.8476337  -2.3819492   1.1581259 ]
  [ 0.29582188 -1.6428163  -1.1351309  -2.3753624   6.571176
   -4.508656   -2.573465   -0.43898493  5.2973313  -1.8500149 ]
  [ 0.94792557 -1.3827156  -2.710929    0.70267767 -4.508656
    6.8080935   0.086738    2.382368   -5.498594   -0.53745675]
  [-1.7730689   0.25873902  6.666974    4.4945393  -2.573465
    0.086738    6.919965   -0.40823123  1.9532615  -0.06429972]
  [ 3.0220997  -1.2621315   0.74811184  1.8476337  -0.43898493
    2.382368   -0.40823123  5.7033362  -2.0585406   0.21069823]
  [-2.5740616  -0.7767093   2.4979637  -2.3819492   5.2973313
   -5.498594    1.9532615  -2.0585406   9.462108   -3.9543934 ]
  [ 2.1073136   4.0993524   1.5011857   1.1581259  -1.8500149
   -0.53745675 -0.06429972  0.21069823 -3.9543934   7.435262  ]]

 [[ 4.6643376   2.6854048   4.0593524   0.1535468  -1.5118827
    3.397957   -2.9920995   0.80777925 -1.7318133   0.82475716]
  [ 2.6854048   7.829547   -0.5087376   4.3746657  -1.4713306
   -4.861691   -6.0652037   0.99772763 -4.4356337   5.25942   ]
  [ 4.0593524  -0.5087376  13.68554    -5.680109   -4.2268295
    8.056314   -6.3996224  -0.8219479  -1.7537458   1.7455587 ]
  [ 0.1535468   4.3746657  -5.680109   15.329586   -1.0279889
   -9.684909   -4.311326   -0.32577878 -8.493642    7.0410876 ]
  [-1.5118827  -1.4713306  -4.2268295  -1.0279889   6.4806786
   -1.5806129   3.722421    1.541739   -0.32345185 -3.0850477 ]
  [ 3.397957   -4.861691    8.056314   -9.684909   -1.5806129
   15.264503    3.3202677  -0.3396074   4.926388   -7.346789  ]
  [-2.9920995  -6.0652037  -6.3996224  -4.311326    3.722421
    3.3202677  11.730729    1.6513809   7.1151867  -8.045567  ]
  [ 0.80777925  0.99772763 -0.8219479  -0.32577878  1.541739
   -0.3396074   1.6513809   4.239041   -0.10079275 -0.07716311]
  [-1.7318133  -4.4356337  -1.7537458  -8.493642   -0.32345185
    4.926388    7.1151867  -0.10079275 12.2900095  -7.07966   ]
  [ 0.82475716  5.25942     1.7455587   7.0410876  -3.0850477
   -7.346789   -8.045567   -0.07716311 -7.07966     9.485415  ]]]
[[[ 3.2763293  -1.4885036  -0.96310735 -1.4039351   0.641888
   -0.3795465   0.9250249  -1.2665155  -0.90407383  0.01106563]
  [ 0.          2.584822    0.9204     -0.8228325  -0.03568195
    1.3950504   0.83006996 -0.599167   -0.12279119 -1.4740415 ]
  [ 0.          0.          1.4903773   0.38091183  0.9942048
   -0.83908963 -0.43472758 -1.7813463  -0.8888414  -0.5626888 ]
  [ 0.          0.          0.          1.9726999   0.12664968
    0.23359051  0.3006057   1.2782736  -1.4336116   0.27203056]
  [ 0.          0.          0.          0.          1.5753931
   -2.2355907   1.1398793   0.4796653   0.9255097   0.5168721 ]
  [ 0.          0.          0.          0.          0.
    1.9982008  -0.8125721   0.22821037  0.14752597  0.5606847 ]
  [ 0.          0.          0.          0.          0.
    0.          1.7990714   0.67006654  0.87395006 -1.4603251 ]
  [ 0.          0.          0.          0.          0.
    0.          0.          1.7880605   1.3292007  -0.657102  ]
  [ 0.          0.          0.          0.          0.
    0.          0.          0.          1.0810157  -0.46530387]
  [ 0.          0.          0.          0.          0.
    0.          0.          0.          0.          0.87528366]]

 [[ 2.4292743  -0.0476496  -0.75327885 -0.03857401  0.12177376
    0.39020938 -0.729876    1.2440339  -1.0596011   0.86746633]
  [ 0.          3.2469115   0.23262091 -1.8222985  -0.50417566
   -0.42012918  0.06897654 -0.37046087 -0.25476483  1.2752694 ]
  [ 0.          0.          3.9448855   1.7044525  -0.2347646
   -0.5879161   1.5465921   0.4490354   0.44590706  0.47098336]
  [ 0.          0.          0.          2.7274456  -1.059335
    0.3498526   0.7171502   0.16688555 -1.3371879   0.9946084 ]
  [ 0.          0.          0.          0.          2.263813
   -2.0034354  -0.58618873 -0.2186787   1.7607771  -0.06559554]
  [ 0.          0.          0.          0.          0.
    1.4133387  -0.08172157  1.067539   -0.6612598  -0.3839536 ]
  [ 0.          0.          0.          0.          0.
    0.          1.768033   -0.18652947  1.3828325  -0.58293706]
  [ 0.          0.          0.          0.          0.
    0.          0.          1.6021042   0.3352417  -0.30378163]
  [ 0.          0.          0.          0.          0.
    0.          0.          0.          0.8516632  -0.96378136]
  [ 0.          0.          0.          0.          0.
    0.          0.          0.          0.          1.5273129 ]]

 [[ 2.1597078   1.2434112   1.8795841   0.0710961  -0.7000404
    1.5733411  -1.3854187   0.37402248 -0.8018739   0.38188368]
  [ 0.          2.5066862  -1.1352971   1.7099324  -0.23971593
   -2.719926   -1.7323902   0.21249725 -1.3717611   1.9087278 ]
  [ 0.          0.          2.977214   -1.3006988  -1.0691853
    0.6755185  -1.9354969  -0.43117726 -0.60590625  1.0730667 ]
  [ 0.          0.          0.          3.272437   -0.5986391
   -1.3039918  -1.1514527  -0.39009395 -2.102138    1.572491  ]
  [ 0.          0.          0.          0.          2.1051443
   -0.5650853  -0.20018531  0.5510192  -1.4820251  -0.12896962]
  [ 0.          0.          0.          0.          0.
    1.707355    0.2816568  -0.1500179  -0.41726547 -0.8804634 ]
  [ 0.          0.          0.          0.          0.
    0.          1.2723097   1.1054928  -0.11362489 -0.07867548]
  [ 0.          0.          0.          0.          0.
    0.          0.          1.4722989   0.19627525  0.32361656]
  [ 0.          0.          0.          0.          0.
    0.          0.          0.          1.5991338  -0.51923645]
  [ 0.          0.          0.          0.          0.
    0.          0.          0.          0.          0.9485597 ]]]
Mismatched elements: 253 / 300 (84.3%)
Max absolute difference among violations: 20.29966
Max relative difference among violations: 237.33156
 ACTUAL: array([[[10.734334, -4.876828, -3.155457, -4.599753,  2.103036,
         -1.243519,  3.030686, -4.149522,  0.      ,  0.      ],
        [-4.876828,  2.244246,  1.433589,  8.771067,  1.423618,...
 DESIRED: array([[[10.734333, -4.876828, -3.155457, -4.599753,  2.103036,
         -1.243519,  3.030686, -4.149522, -2.962044,  0.036255],
        [-4.876828,  8.896948,  3.812659, -0.037113, -1.047684,...
failed, 47742 passed, 498 skipped, 33 xfailed, 5 xpassed, 6 warnings in 14045.71s (3:54:05)
```

</details>
